sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionRoundRobin.ts

```diff
@@ -0,0 +1,146 @@
+/*
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT Licence.
+
+  This sample demonstrates how you can continually read through all the available
+  sessions in a Service Bus queue or subscription.
+
+  Run the sendMessages sample with some session ids before running this sample.
+*/
+
+import {
+  ServiceBusClient,
+  delay,
+  ReceivedMessageWithLock,
+  SessionReceiver,
+  MessagingError
+} from "@azure/service-bus";
+import * as dotenv from "dotenv";
+import { env } from "process";
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+
+dotenv.config();
+
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+
+// NOTE: this sample uses a queue but would also work a session enabled subscription.
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutMs = 3 * 1000;
+const delayWhenNoSessionsAvailableMs = 5 * 1000;
+
+// this can be used control when the round-robin processing will terminate.
+const abortController = new AbortController();
+
+// called just before we start processing the first message of a session
```

Comment:
> called just before ...

I worry that this phrasing gives the impression that it gets passed to the service library and that the library will call it. Especially because `processMessage` and `processError` which do get passed to the library have the same phrasing. Can we rephrase that this is something that the user needs to call?

>  .. just before we start processing the first message of a session

Should we make it clear that this is called before we start receiving at all? I wonder if the current text makes it feel that we receive the first message, but before we pass it to `processMessage`, we call the `processInitialize`

My recommendation:

> Helper function for you to call once a session receiver is initialized, but before receiving any messages

And then for `processMessage`

> Helper function to be passed to the subscribe() method on the receiver to be called for each received message

Reply:
I'm not sure I understand the concern. The user can easily see how these functions are used.

Also, the rewording doesn't seem accurate either - the user isn't expected to call these for the sample at all - that's all handled for them.

Reply:
> that's all handled for them

handled by the sample vs handled by the library: that is the distinction I want to make

`processMessage` and `processError` are called by the library
`processInitialize` and `processClose` are called by the sample. And the sample for all means and purposes is user code, not library code

Reply:
My point is - what problem are you trying to solve here? When we had an interface I could see the issue but the functions and their call points are clearly documented and viewable.

What confusion do you think is going to happen on the user's side? That they'll go to the main service package and try to use these methods there?

Reply:
After offline discussion with @ramya-rao-a we agreed to modify the wording. I've also changed the names of the functions to better distinguish them from the actual handler methods that are passed into `subscribe`.